botocore/signers.py

```diff
@@ -123,6 +113,36 @@ def sign(self, operation_name, request):
                                             signature_version)
             signer.add_auth(request=request)
 
+    def choose_signer(self, operation_name, suffix=''):
+        """
+        Allow overriding the signature version via the choose-signer event.
+        A value of `botocore.UNSIGNED` means no signing will be performed.
+
+        :param operation_name: The operation to sign.
+        :param suffix: A suffix that needs to be on all valid signers. This
```

Comment:
This arg doesn't seem to make sense as a public interface.  First, suffix is completely implementation specific on how we decided to name signature versions.  Secondly, seems like it should be the callers responsibility to deal with interpreting signature versions if they care.